src/promptflow-devkit/tests/sdk_cli_test/unittests/test_trace.py

```diff
@@ -211,3 +213,67 @@ def _validate_invalid_params(kwargs: Dict):
         _validate_invalid_params({"run": str(uuid.uuid4()), "started_before": datetime.datetime.now().isoformat()})
         _validate_invalid_params({"collection": TRACE_DEFAULT_COLLECTION})
         _validate_invalid_params({"collection": str(uuid.uuid4()), "started_before": "invalid isoformat"})
+
+
+@pytest.mark.unittest
+@pytest.mark.sdk_test
+class TestWorkspaceKindLocalCache:
+    def test_no_cache(self):
+        mock_sub, mock_rg, mock_ws = str(uuid.uuid4()), str(uuid.uuid4()), str(uuid.uuid4())
+        ws_local_cache = WorkspaceKindLocalCache(
+            subscription_id=mock_sub, resource_group_name=mock_rg, workspace_name=mock_ws
+        )
+        assert not ws_local_cache.is_cache_exists
+        # mock `WorkspaceKindLocalCache._get_workspace_kind_from_azure`
+        mock_kind = str(uuid.uuid4())
+        with patch(
+            "promptflow._sdk._tracing_utils.WorkspaceKindLocalCache._get_workspace_kind_from_azure"
+        ) as mock_get_kind:
+            mock_get_kind.return_value = mock_kind
+            assert ws_local_cache.get_kind() == mock_kind
+
+    def test_valid_cache(self):
+        mock_sub, mock_rg, mock_ws = str(uuid.uuid4()), str(uuid.uuid4()), str(uuid.uuid4())
+        # manually create a valid local cache
+        mock_kind = str(uuid.uuid4())
+        with open(HOME_PROMPT_FLOW_DIR / "tracing" / f"{mock_sub}_{mock_rg}_{mock_ws}.json", "w") as f:
+            cache = {
+                WorkspaceKindLocalCache.SUBSCRIPTION_ID: mock_sub,
+                WorkspaceKindLocalCache.RESOURCE_GROUP_NAME: mock_rg,
+                WorkspaceKindLocalCache.WORKSPACE_NAME: mock_ws,
+                WorkspaceKindLocalCache.KIND: mock_kind,
+                WorkspaceKindLocalCache.TIMESTAMP: datetime.datetime.now().isoformat(),
+            }
+            f.write(json.dumps(cache))
+        ws_local_cache = WorkspaceKindLocalCache(
+            subscription_id=mock_sub, resource_group_name=mock_rg, workspace_name=mock_ws
+        )
+        assert ws_local_cache.is_cache_exists is True
+        assert not ws_local_cache.is_expired
+        assert ws_local_cache.get_kind() == mock_kind
+
+    def test_expired_cache(self):
+        mock_sub, mock_rg, mock_ws = str(uuid.uuid4()), str(uuid.uuid4()), str(uuid.uuid4())
```

Comment:
change the word mock to something else...
In these tests.

Reply:
updated to remove these "mock" usage